ddtrace/ext/sql.py

```diff
@@ -23,14 +27,54 @@ def normalize_vendor(vendor):
         return vendor
 
 
-def parse_pg_dsn(dsn):
+def _dd_parse_pg_dsn(dsn):
     # type: (str) -> Dict[str, str]
     """
     Return a dictionary of the components of a postgres DSN.
     >>> parse_pg_dsn('user=dog port=1543 dbname=dogdata')
     {'user':'dog', 'port':'1543', 'dbname':'dogdata'}
     """
-    return dict(_.split("=", 1) for _ in dsn.split())
+    dsn_dict = dict()
+    try:
+        # dsn strings with the following format raise a TypeError:
+        # dsn="db=moon user=ears options='-c statement_timeout=1000 -c lock_timeout=250'"
+        dsn_dict = dict(_.split("=", 1) for _ in dsn.split())
+    except Exception:
+        log.debug("Failed to parse dsn string", exc_info=True)
+    return dsn_dict
+
+
+# Do not import from psycopg directly! This reference will be updated at runtime if we can
+# import psycopg 2 or psycopg 3 and use a better implementation.
```

Comment:
```suggestion
# Do not import from psycopg directly! This reference will be updated at runtime to use
# a better implementation that is provided by the psycopg library.
# This is done to avoid circular imports.
```